### ISSUE TITLE
[Research] Music-Specific EEG Features for Genre-Aware Emotion Classification

### ISSUE THREAD

**Issue by LakshmiSravyaVedantham** [LakshmiSravyaVedantham/DreamAnalysis#136]
## Research Topic
Music-induced emotion recognition using EEG features that are distinct from general emotion classification — temporal dynamics of emotional transitions during music, genre-specific neural signatures, and alpha entrainment to musical tempo.

## Paper References

1. **"Music emotion recognition based on temporal convolutional attention network using EEG"** — Qiao et al., Frontiers in Human Neuroscience, April 2024. Built CNN-SA-BiLSTM model for music emotion classification from EEG. Created custom dataset with 4 music emotion categories (fear, happiness, calm, sadness). Achieved **93.45% valence and 96.36% arousal accuracy** (within-subject). Key insight: self-attention mechanism captures temporal transitions between emotions during a musical piece — emotions are NOT static during music, they flow. Validated on DEAP dataset (87-88% accuracy).

2. **"A review of artificial intelligence methods enabled music-evoked EEG emotion recognition and their applications"** — PMC, September 2024. Comprehensive review of AI for music-evoked EEG emotion recognition. Key findings: (a) music-evoked emotions show **distinct EEG patterns** from visual/social emotion induction, (b) frontal alpha asymmetry is the strongest valence marker for music-induced emotion specifically, (c) temporal channels (TP9/TP10 on Muse 2) are MORE relevant for music emotion than general emotion — proximity to auditory cortex matters, (d) frequency-band contributions differ: theta is more important for music (absorption/flow) than for general emotion.

3. **"Music emotion interaction system based on brain-computer interface: a hybrid deep learning and transfer learning framework"** — CCF Transactions on Pervasive Computing and Interaction, 2025. Hybrid deep learning + transfer learning framework for music-EEG emotion BCI. Uses transfer learning to adapt from lab datasets to real-world consumer EEG. Demonstrates that **pre-training on DEAP then fine-tuning on consumer data** bridges the domain gap effectively.

4. **"An EEG-Based Analysis of the Effects of Different Music Genres on Driving Stress"** — Frontiers in Human Neuroscience, 2025. Measured genre-specific EEG effects during driving. Classical music **reduced frontal beta power** (stress marker). Heavy metal **increased beta and gamma power**. Pop music had intermediate effects. Direct evidence for genre-specific neural signatures detectable from frontal EEG.

5. **"Music-Evoked Emotions Classification Using Vision Transformer in EEG Signals"** — Scientific Reports, 2024. Applied Vision Transformer (ViT) to EEG spectrograms for music emotion classification. Novel approach: converts EEG signal processing into a computer vision problem by treating time-frequency representations as images. Competitive accuracy with much simpler feature engineering.

## Application to NeuralDreamWorkshop

This is distinct from the existing general emotion classifier — music-induced emotion uses different feature weights and has unique characteristics:

### Why music emotion is different from general emotion:
1. **Temporal dynamics**: emotions transition continuously during music (not static windows)
2. **Temporal channels matter more**: TP9/TP10 (near auditory cortex) are primary for music, secondary for general emotion
3. **Theta is more important**: musical absorption increases theta (whereas general emotion theta = drowsiness)
4. **Genre creates predictable baselines**: classical → low beta, metal → high beta
5. **Alpha entrainment**: alpha oscillations can synchronize to musical tempo (unique to auditory stimuli)

### What already exists:
- Full 4-channel EEG pipeline including TP9/TP10 (temporal channels)
- Alpha, theta, beta power extraction
- FAA computation (valence marker)
- Differential entropy per band
- Existing emotion classifier with 6 classes

### What this adds:
- Music-specific feature extraction (alpha entrainment, temporal channel weighting)
- Emotion transition detection (flowing between states during music)
- Genre-specific baseline normalization
- Integration with audio metadata for context-aware classification

### New files:
- `ml/models/music_emotion_classifier.py` — MusicEmotionClassifier class
- `ml/processing/music_eeg_features.py` — music-specific EEG feature extraction
- `client/src/pages/music-emotion.tsx` — music listening session with EEG overlay

### Key classes/methods:
```python
class MusicEmotionClassifier:
    """Classifies emotional response to music with music-specific features."""
    
    def __init__(self):
        self.base_emotion_classifier = None  # reuse existing
        self.temporal_weight = 0.4  # TP9/TP10 weight (higher than general emotion)
        self.frontal_weight = 0.6   # AF7/AF8 weight
    
    def extract_music_features(self, eeg_4ch: np.ndarray, fs: int = 256,
                                tempo_bpm: float = None) -> dict:
        """Extract music-specific EEG features.
        Returns:
        - temporal_alpha: TP9/TP10 alpha (auditory processing)
        - temporal_theta: TP9/TP10 theta (musical absorption)
        - alpha_entrainment: correlation of alpha power with music tempo
        - frontal_beta_suppression: stress reduction during pleasant music
        - emotion_transition_rate: how rapidly valence/arousal are changing
        - faa_dynamic: FAA computed with sliding window (captures transitions)
        - genre_features: beta/gamma ratio (metal high, classical low)"""
    
    def classify_with_transitions(self, eeg_4ch: np.ndarray, fs: int = 256,
                                   window_sec: float = 4.0,
                                   hop_sec: float = 1.0) -> list[dict]:
        """Classify emotion at each hop, returning emotion trajectory.
        Returns list of: {time_sec, valence, arousal, emotion, transition_detected}"""
    
    def compute_engagement_score(self, eeg_4ch: np.ndarray, fs: int = 256) -> float:
        """Musical engagement: theta increase + alpha decrease + low beta.
        Similar to flow state but with temporal channel emphasis.
        Returns: 0.0-1.0 engagement score"""
```

### Key features (unique to music emotion):
- **Alpha entrainment index**: cross-correlation between alpha power envelope and beat timing
- **Temporal theta increase**: TP9/TP10 theta during musical absorption (distinct from drowsiness)
- **Beta suppression ratio**: frontal beta reduction during calming music (stress relief marker)
- **Emotion transition rate**: derivative of valence/arousal over sliding windows
- **Genre baseline**: expected beta/gamma levels per genre (learned or preset)

### API endpoints:
- `POST /music-emotion/analyze` — classify emotion during music listening
- `POST /music-emotion/start-session` — start music listening session with EEG tracking
- `GET /music-emotion/trajectory` — emotion trajectory over session
- `GET /music-emotion/engagement` — musical engagement score

## Expected Accuracy/Impact
- Music emotion classification (within-subject): **85-93%** valence, **90-96%** arousal (from TCAN paper)
- Cross-subject (realistic for Muse 2): **65-75%** (same domain gap as general emotion)
- With personalization (3-5 sessions): **75-85%**
- TP9/TP10 utilization: temporal channels become PRIMARY features (currently underused in general emotion)
- Alpha entrainment: detectable but subtle (~5-10% power modulation at beat frequency)
- Unique differentiator: emotion trajectory visualization during music — not available in any consumer BCI app

## Complexity Rating: EASY-MEDIUM
- Feature extraction builds on existing pipeline — adds temporal channel weighting + entrainment
- Quick-win: reweight existing emotion classifier for music context (1 day)
- Medium work: add transition detection + entrainment analysis (3-5 days)
- No new training data required for v1 — feature-based approach
- Audio metadata (tempo, genre) enhances accuracy but is optional
- Main UI work: emotion trajectory visualization over time
- Estimated implementation: 1 week
